src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -7,6 +7,17 @@ namespace Datadog.Trace.Agent.MessagePack
 {
     internal class SpanMessagePackFormatter : IMessagePackFormatter<Span>
     {
+        private static byte[] _traceIdBytes = StringEncoding.UTF8.GetBytes("trace_id");
+        private static byte[] _spanIdBytes = StringEncoding.UTF8.GetBytes("span_id");
+        private static byte[] _nameBytes = StringEncoding.UTF8.GetBytes("name");
+        private static byte[] _resourceBytes = StringEncoding.UTF8.GetBytes("resource");
+        private static byte[] _serviceBytes = StringEncoding.UTF8.GetBytes("service");
+        private static byte[] _typeBytes = StringEncoding.UTF8.GetBytes("type");
+        private static byte[] _startBytes = StringEncoding.UTF8.GetBytes("start");
+        private static byte[] _durationBytes = StringEncoding.UTF8.GetBytes("duration");
+        private static byte[] _parentIdBytes = StringEncoding.UTF8.GetBytes("parent_id");
+        private static byte[] _errorBytes = StringEncoding.UTF8.GetBytes("error");
```

Comment:
Should we make these not `static`? Did we benchmark in .NET Framework loading `Datadog.Trace.dll` domain-neutral?

>_Access to static data and methods is slower for domain-neutral assemblies_ because of the need to isolate assemblies. Each application domain that accesses the assembly must have a separate copy of the static data, to prevent references to objects in static fields from crossing domain boundaries. As a result, the runtime contains additional logic to direct a caller to the appropriate copy of the static data or method. This extra logic slows down the call.

source: https://docs.microsoft.com/en-us/dotnet/framework/app-domains/application-domains

Reply:
Thanks for this, as we talked, I'll run some benchmarks for curiosity about the performance impact on domain neutral assemblies. But it doesn't make any harm to change it to instance fields in the `SpanMessagePackFormatter` file if we double check we use a single instance.